### ISSUE TITLE
💅 noSolidDestructuredProps doesn't work

### ISSUE THREAD

**Issue by yunics-highfield** [biomejs/biome#10742]
### Environment information

<details>

```bash

Paste output here

```

</details>

### Rule name

noSolidDestructuredProps

### Playground link

https://biomejs.dev/playground/?tab=formatter&pane=Diagnostics&code=aQBtAHAAbwByAHQAIAB7ACAAcgBlAG4AZABlAHIAIAB9ACAAZgByAG8AbQAgACcAcwBvAGwAaQBkAC0AagBzAC8AdwBlAGIAJwA7AAoACgB0AHkAcABlACAAUAByAG8AcABzACAAPQAgAHsACgAgACAAZABhAHQAYQA6ACAAbgB1AG0AYgBlAHIAOwAKAH0ACgAKAC8ALwAgAG4AbwBTAG8AbABpAGQARABlAHMAdAByAHUAYwB0AHUAcgBlAGQAUAByAG8AcABzACAAZABvAGUAcwBuACcAdAAgAHcAbwByAGsALgAKAGYAdQBuAGMAdABpAG8AbgAgAE0AeQBDAG8AbQBwAG8AbgBlAG4AdAAoAHsAIABkAGEAdABhACAAfQA6ACAAUAByAG8AcABzACkAIAB7AAoAIAAgAHIAZQB0AHUAcgBuACAAPABkAGkAdgA%2BAHsAIABkAGEAdABhACAAfQA8AC8AZABpAHYAPgAKAH0ACgAKAGYAdQBuAGMAdABpAG8AbgAgAEEAcABwACgAKQAgAHsACgAgACAAcgBlAHQAdQByAG4AIAA8AE0AeQBDAG8AbQBwAG8AbgBlAG4AdAAgAC8APgAKAH0ACgAKAHIAZQBuAGQAZQByACgAKAApACAAPQA%2BACAAPABBAHAAcAAgAC8APgApADsA&ruleDomains.solid=all

### Expected result

noSolidDestructuredProps rule should work.

I'm worried about package name as well. `solid-js` is right.
https://github.com/biomejs/biome/blob/ffe7d33212f7cb89162bb222ce29e3468b75488d/crates/biome_analyze/src/rule.rs#L669

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Mmmh, weird. The rule actually catches props that are used inside component props. If `data` is used in `foo={data}`, you'll see the diagnostic. 

Does the eslint rule catch this case  @yunics-highfield  ?

**Comment by yunics-highfield** [comment]
@ematipico  I created a sample SolidJS app for ESLint. The app shows the error.

<img width="1147" height="381" alt="Image" src="https://github.com/user-attachments/assets/02e617e8-7238-4440-849e-d39295b1c782" />
